runner.py

```diff
@@ -0,0 +1,495 @@
+    flags : dict[str, Union[str, None]]
+        The flags which have been passed to the installation script.
+    
+    Returns
+    -------
+    bool
+        True if the flags were valid otherwise false.
+    Union[Exception, None]
+        None if the flags were valid otherwise a corresponding error.
+    '''
+
+    if any(map(lambda opt: opt is not None, flags.values())):
+        passed_arg_pairs = filter(lambda pair: pair[1] is not None, flags.items())
+        passed_args = ', '.join(map(lambda pair: pair[0], passed_arg_pairs))
+        return False, Exception(f'Too many arguments: Do not specify {passed_args} when ' +
+        'using the start flag.')
+
+    return True, None
+
+
+def get_flags() -> dict[str, str]:
+    '''
+    Retrieves the flags given by the user and returns them as a dictionary.
+
+    Example Output:
+    {
+        'token': 'XXXXXXXXXXXXXXXXXXXXXXXXXXXXX',
+        'url': 'https://github.com/diku-dk/futhark',
+        'name': 'runner',
+        'labels': 'cuda,opencl,multicore'
+    }
+    
+    Returns
+    -------
+    dict[str, str]
+        The flags given by the user as a dictionary.
+    '''
+
+    parser = optparse.OptionParser()
+    parser.add_option('-t', '--token', dest='token', type='string',
+                      help='the runner TOKEN.', metavar='TOKEN')
+    parser.add_option('-u', '--url', dest='url', type='string',
+                      help='the URL to the repository the runner is used for. ' + 
+                      'Defaults to "https://github.com/diku-dk/futhark"', metavar='URL')
+    parser.add_option('-n', '--name', dest='name', type='string',
+                      help='the NAME of the runner.', metavar='NAME')
+    parser.add_option('-l', '--labels', dest='labels', type='string',
+                      help='the LABELS the runner should have.', metavar='LABELS')
+    parser.add_option('-s', '--start', action="store_true", dest="start", default=False,
+                      help='If the runner should START.', metavar='START')
+    parser.add_option('-r', '--remove', dest="remove",
+                      help='the token belonging to the runner that will be REMOVEd.', metavar='REMOVE')
+    (flags, _) = parser.parse_args()
+    flags = flags.__dict__
+    
+    if flags.get('start'):
+        start = flags.pop('start')
+
+        is_valid, error = is_all_none_flags(flags)
+        if not is_valid:
+            raise error
+        
+        flags = {'start': start}
+        return flags
+    
+    flags.pop('start')
+
+    if flags.get('remove') is not None:
+        remove = flags.pop('remove')
+
+        is_valid, error = is_all_none_flags(flags)
+        if not is_valid:
+            raise error
+
+        flags = {'remove': remove}
+        return flags
+    
+    remove = flags.pop('remove')
+    
+    if flags.get('url') is None:
+        flags['url'] = 'https://github.com/diku-dk/futhark'
+
+    hostname = socket.gethostname()
+    settings = DEFAULT_SETTINGS.get(hostname)
```

Comment:
Won't this fail with some key-not-found exception if the hostname is not known?  That is poor UX.  Also, I think the script should explicitly mention the name and label it extracts from the defaults, if any.

Reply:
I believe it should not throw an error since [get](https://docs.python.org/3/library/stdtypes.html#dict.get) returns None if the key is not known. Also I will make it so the script explicitly mentions when default settings are being used.